### ISSUE TITLE
Bug in Tuner module (Random search & Grid search)

### ISSUE THREAD

**Issue by Yvnminc** [manujosephv/pytorch_tabular#388]
**Describe the bug**
It seems the tuner module does not correctly copy the model configuration. It does not update the right parameters in the search grid. It just used the default param.

**To Reproduce**
Try the following code:

from pytorch_tabular.tabular_model_tuner import TabularModelTuner

params = {
    "model_config__num_attn_blocks": range(1, 5),
    "model_config__num_heads": [4,8],
    "model_config__ff_hidden_multiplier": range(1, 5),
    'model_config__learning_rate': [1e-2, 1e-3],
}

tuner = TabularModelTuner(
            data_config = data_config,
            model_config = model_config,
            optimizer_config = optimizer_config,
            trainer_config = trainer_config
        )

train, test = train_test_split(data, test_size=0.2)
train, val = train_test_split(train, test_size=0.2)

X_y_train = train
X_y_val = val

result = tuner.tune(
    train=X_y_train,
    validation=X_y_val,
    search_space=params,
    strategy="random_search",
    n_trials=1,
    metric = "mean_squared_error",
    mode="min",
    progress_bar=True,
    verbose=False # Make True if you want to log metrics and params in each iteration
)

result.trials_df.sort_values(by= "mean_squared_error", ascending= True)

**Expected behavior**
It returns the result of the param in the search grid.

**Screenshots**
I posted two screenshots of unreasonable parameters that could not converge but with good performance.

<img width="1161" alt="Unexpected output1" src="https://github.com/manujosephv/pytorch_tabular/assets/111160898/e42622dd-8c65-4894-8dd0-8789cbfbe6a2">

<img width="1078" alt="Unexpected output2" src="https://github.com/manujosephv/pytorch_tabular/assets/111160898/d12487ec-6f66-4b85-9a94-1ad2693d1b04">

**Desktop (please complete the following information):**

- OS: MacOS
- Version: 14.1

**Additional context**
I studied your amazing paper: GANDALF: Gated Adaptive Network for Deep Automated Learning of Features for Tabular Data. The model performs SOTA on my dataset. I noticed you utilise the BO tuning algorithm. I did not find the function BO implemented in PyTorch Tabular; is it still under development?

**Comment by manujosephv** [comment]
@Yvnminc First of all, thank you. It makes me happy to see my model performing SOTA on any dataset :D 

And yes, this does look weird. Then again, the tuner module seemed to have some bugs. @ProgramadorArtificial Do you think this is one of the bugs you fixed recently?

@Yvnminc In the meanwhile, can you retry this experiment with latest code from `main`. There were a few fixed merged recently, just wanted to figure out if it fixed this or not.

And the BO tuning algorithm - is in the roadmap. For GANDALF experiments, I wrote my own code to perform BO using `optuna`. But I haven't integrated it to PyTorch Tabular. It's an item that is currently open for contribution.

**Comment by ProgramadorArtificial** [comment]
The parameters that are not changing are already corrected and were merged into the main two days ago (#384). If you update your library, it should work fine. Except changes in "TrainerConfig" and "OptimizerConfig" (which is not your example case), these I am working on it yet (#387).
